Update to operator-sdk 0.15 (#109)
codeready-toolchain/api: scripts/README.adoc

The `olm-catalog-generate.sh` script (re)generates OLM catalog files in their respective operator repositories/directories.
It generates:

* The operator bundle that is used as the base template for all next releases.
* Some "hack" files, including:
** `deploy_csv.yaml` that helps in deploying the current version of the operator bundle to the cluster. It's the same file as is generated using <<generate-deploy-hack,generate-deploy-hack.sh>> script.
** `install_operator.yaml` that installs the operator using `OperatorGroup` and `Subscription` resources.

Under the hood, it runs the `operator-sdk` command with specific parameters.

> Based on the fact if the operator repository contains CRD types (if it contains `pkg/apis/toolchain/v1alpha1` directory) it decides where the `operator-sdk` command will be executed.

Reviewer: ```suggestion
If the operator repository contains CRD types (i.e., contains the `pkg/apis/toolchain/v1alpha1` directory) it decides where the `operator-sdk` command will be executed.
```

Reviewer: @MatousJobanek in that case e.g., can be i.e.,

> * If the repository contains the CRD types (in case of toolchain-operator) then the command is executed inside of the operator repo.

Reviewer: ```suggestion
* If the repository contains the CRD types (e.g., toolchain-operator) then the command is executed inside of the operator repo.
```

Reviewer 2: same as the previous - it's not an example, but the specific case

Reviewer: @MatousJobanek in that case e.g., can be `i.e.,`

> * If the repository doesn't contain the CRD types (in case of hosted toolchain operators) then the command is executed inside of codeready-toolchain/api repo. To do that, the script:

Reviewer: ```suggestion
* If the repository doesn't contain the CRD types (e.g., hosted toolchain operators) then the command is executed inside of the codeready-toolchain/api repo. To do that, the script:
```

Reviewer 2: it's not an example - it's really the only case

Reviewer: @MatousJobanek in that case e.g., can be i.e.,

Reviewer 2: @tinakurian could you please propose the change as a suggestion?
